src/lib/core/components/computations/plugins/abundance.tsx

```diff
@@ -1,25 +1,86 @@
 /** @jsxImportSource @emotion/react */
-import { jsx } from '@emotion/react';
-import { useRouteMatch } from 'react-router-dom';
-import { useHistory } from 'react-router';
 import { useStudyMetadata } from '../../..';
 import { useCollectionVariables } from '../../../hooks/study';
 import { VariableDescriptor } from '../../../types/variable';
+import { StudyEntity } from '../../../types/study';
 import { boxplotVisualization } from '../../visualizations/implementations/BoxplotVisualization';
 import { scatterplotVisualization } from '../../visualizations/implementations/ScatterplotVisualization';
 import { ComputationConfigProps, ComputationPlugin } from '../Types';
 import { H6 } from '@veupathdb/coreui';
 import { isEqual } from 'lodash';
-import { createComputation } from '../Utils';
+import { assertComputationWithConfig, useConfigChangeHandler } from '../Utils';
+import { findCollections } from '../../../utils/study-metadata';
+import * as t from 'io-ts';
+import { Computation } from '../../../types/visualization';
+
+export type AbundanceConfig = t.TypeOf<typeof AbundanceConfig>;
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export const AbundanceConfig = t.type({
+  name: t.string,
+  collectionVariable: VariableDescriptor,
+  rankingMethod: t.string,
+});
 
 export const plugin: ComputationPlugin = {
   configurationComponent: AbundanceConfiguration,
+  configurationDescriptionComponent: AbundanceConfigDescriptionComponent,
   visualizationTypes: {
     boxplot: boxplotVisualization,
     scatterplot: scatterplotVisualization,
   },
+  createDefaultComputationSpec: createDefaultComputationSpec,
 };
 
+function AbundanceConfigDescriptionComponent({
```

Comment:
This is great. Leaves us lots of room to personalize the configuration description per app, which is going to be really helpful